web/src/components/organisms/Settings/Integration/hooks.ts

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useMemo, useState } from "react";
+
+import {
+  Integration,
+  IntegrationMember,
+  Role,
+} from "@reearth-cms/components/molecules/Integration/types";
+import {
+  useGetMeQuery,
+  useAddIntegrationToWorkspaceMutation,
+  Role as GQLRole,
+  useUpdateIntegrationOfWorkspaceMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+
+export default (workspaceId?: string) => {
+  const [selectedConnectionModalIntegration, SetSelectedConnectionModalIntegration] =
```

Comment:
This name is confusing. Is it just a selectedIntegration? Or is it actually "an integration that is a connection modal that is selected"??

Reply:
@KaWaite , yes it is related only to the modal, because there is an `IntegrationMember` selected as well.
To avoid confusion I named it like this.

Reply:
@issmail-basel Is there any way you can move this inside of the modal itself, it can handle it and keep things simpler? If you move this in to the modal, you can have a selectedIntegration prop that you pass whichever integration has been selected for modal open. Then, on close just clear this field (with any other fields you need to clear).
What do you think?

Reply:
@KaWaite , moved the state inside the modal itself, the code is more readable now in my opinion.

https://github.com/reearth/reearth-cms/blob/0d2d4093be036c6fad8fb94a23aabf14657cf87e/web/src/components/molecules/Integration/IntegrationConnectModal/index.tsx#L19-L27